raw/fetch.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import datetime
+import json
+import os
+import re
+import subprocess
+from google.cloud import secretmanager
+from typing import Tuple
+
+DIRECTORY = "/weather/config_files"
+FIELD_NAME = "date"
+PROJECT = os.environ.get("PROJECT")
+REGION = os.environ.get("REGION")
+BUCKET = os.environ.get("BUCKET")
+SDK_CONTAINER_IMAGE = os.environ.get("SDK_CONTAINER_IMAGE")
+MANIFEST_LOCATION = os.environ.get("MANIFEST_LOCATION")
+API_KEY_PATTERN = re.compile(r'^API_KEY_\d+$')
+API_KEY_LIST = []
+
+
+def new_config_file(config_file: str, field_name: str, additional_content: str,
```

Comment:
Please add unit tests for this script.

Reply:
Code updated as per your suggestion. Thanks for the input!